src/v/kafka/server/usage_aggregator.cc

```diff
@@ -181,6 +181,21 @@ usage_aggregator<clock_type>::usage_aggregator(
     _buckets[_current_window].reset(ss::lowres_system_clock::now());
 }
 
+template<typename clock_type>
+void usage_aggregator<clock_type>::rearm_window_timer() {
+    /// Calculate the next time the timer should fire, to adjust for skew ensure
+    /// the timer always fires at the top of the interval, weather it be hour,
+    /// minute, second, etc.
+    const auto now = clock_type::now();
+    const auto delta = std::chrono::seconds(
+      epoch_time_secs(now) % _usage_window_width_interval.count());
```

Comment:
nit: think about making a note that this only works because unix epoch is aligned to an hour boundary.